File: Data Structures & Algorithms/is-palindrome/submission-0.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        result = True        
        s = re.sub(r'[^a-zA-Z0-9]','',s).lower()
        right = 0
        left = len(s) -1


        while(right < left):
            if s[right] != s[left] :
                result = False
                break

            right+=1
            left-=1
        
        return result
        
        

File: Data Structures & Algorithms/palindrome-partitioning/submission-0.java
class Solution {
    public List<List<String>> partition(String s) {
        List<List<String>> result = new ArrayList<>();
        helper(result , new ArrayList<>() , s , 0);
        return result;
    }

    private void helper(List<List<String>> result , List<String> tempList , String s , int index){
        if (index == s.length()){
            result.add(new ArrayList<>(tempList));
            return;
        }

        for (int i=index;i<s.length();i++){
            String subStr = s.substring(index , i+1);
            if(isPalindrome(subStr)){
                tempList.add(subStr);
                helper(result , tempList , s , i+1);
                tempList.remove(tempList.size()-1);
            }
        }
    }

    private boolean isPalindrome(String s){
        if(s.length()==0){
            return true;
        }

        int left =0;
        int right = s.length()-1;
        while(right > left){
            if(s.charAt(left) != s.charAt(right)){
                return false;
            }
            right --;
            left++;
        }

        return true;
    }
}
